run/helm/snap/templates/daemonset.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: snap
+  namespace: {{ .Release.Namespace }}
+  labels:
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+spec:
+  template:
+    metadata:
+      name: snap
+      labels:
+        daemon: snapteld
+    spec:
+      containers:
+      - name: {{ .Chart.Name }}
+        image: "{{ .Values.image.repo }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        securityContext:
+          privileged: {{ .Values.privileged }}
+        ports:
+        - containerPort: 8181
+        volumeMounts:
+        - name: autoload
+          mountPath: {{ .Values.config.autodiscover_path }}
+        - name: config
+          mountPath: /etc/snap
+        {{- range $key, $val := .Values.mounts }}
+        - name: {{ $key | replace "/" "" | replace "." "" }}
+          mountPath: {{ $val | quote }}
+        {{- end }}
+      volumes:
+      - name: autoload
+        hostPath:
```

Comment:
I think that we should aim to have autoload occur in a way that doesn't alter the host node.  What about using something with a persistent volume claim since they allow claiming storage without fully knowing the details of a particular enviroment?